daffodil-cli/src/test/scala/org/apache/daffodil/cli/cliTest/TestCLIParsing.scala

```diff
@@ -938,8 +938,8 @@ class TestCLIParsing {
     )
     runCLI(args"parse -s $schema") { cli =>
       cli.sendLine("0", inputDone = true)
-      cli.expectErr("Unexpected exception in layer transformer 'buggy'")
-    }(ExitCode.LayerExecutionError)
```

Comment:
An abort is not caught. 

A RuntimeException converts to a LayerRuntimeException, but that then propagates to top level. I.e., isn't caught.

Any unexpected exception becomes a processingError carrying a LayerUnexpectedException carrying the exception. It's a processing error because variables can be populated from expressions, and expressions values can come from the infoset, which could be full of gibberish since the parser could be speculating down a bad path. That's why anything about the variables causes a processing error. 

Consider the gzip layer, and the parser speculating down a path which does not have gzip data. The gzip layer is going to fail somehow and throw an IOException mostlikely but it could throw NPE, or any of a variety of things. 

A layer author could, if they choose, call runtimeSchemaDefinitonError from their layer code. Or they can throw Error (which is at Throwable, but not an Exception. Error will not be caught.)

Reply:
Sounds like our buggy layer test isn't really considered buggy then with the new API? It's just a layer that does speculative parsing that might fail? Maybe it should be changed to throw an NPE or some other runime exception that would trigger the fatal error code and the CLI can output a message about?

Reply:
Yeah, I think Buggy layer is only "buggy" because it never does anything but throw an error. 
The test has no speculative parsing, so the error is not suppressed in this case and the test shows that the CLI exposes the error in this situation.